examples/onnxruntime/pytorch/text-classification/run_glue.py

```diff
@@ -563,18 +566,27 @@ def compute_metrics(p: EvalPrediction):
     else:
         data_collator = None
 
-    if not optim_args.quantize and not optim_args.optimize:
-        raise ValueError("`optimize` and `quantize` are both set to False.")
+    # Raise unsupported situations
+    if not (optim_args.quantize or optim_args.optimize or data_args.ort_train):
+        raise ValueError("None of `onnxruntime training`, `optimize` or `quantize` is enbaled.")
 
-    if not training_args.do_eval:
+    if (optim_args.quantize or optim_args.optimize) and not training_args.do_eval:
         raise ValueError(
             "`do_eval` must be set to True in order to compare the evaluation results between the original and the "
             "optimized models."
         )
 
-    # Initialize our Trainer
-    # TODO : Replace Trainer by our ORTTrainer
-    trainer = Trainer(
+    if (optim_args.quantize or optim_args.optimize) and data_args.ort_train:
+        raise ValueError(
+            "`onnxruntime training` is not supported when the quantization or the graph optimization is enabled."
+        )
+    elif optim_args.quantize and optim_args.optimize:
+        raise ValueError("`quantize` is not supported when the graph optimization is enabled.")
```

Comment:
Very thoughful but I think we can remove it as technically `optim_args.quantize` and `optim_args.optimize` can be both set to True (as long as `ORTOptimizer` and  `ORTQuantizer` don't interefere with each other). ONNX Runtime can still do some basic [optimization ](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/tools/quantization/quantize.py#L30) before quantization, for that `optimize_model` of `ORTOptimizer` must be set to True and [`ORTQuantizer`](https://github.com/huggingface/optimum/blob/main/optimum/onnxruntime/quantization.py#L265)  will take care of the rest

Reply:
Sure, I will remove that!